CactbotOverlay/CactbotOverlay.cs

```diff
@@ -681,7 +681,9 @@ public CactbotOverlay(CactbotOverlayConfig config)
         DispatchToJS(new JSEvents.SendSaveData(Config.OverlayData));
       } else if (obj.ContainsKey("setSaveData")) {
         Config.OverlayData = obj["setSaveData"];
-      }
+      } else if (obj.ContainsKey("getUserLocation")) {
```

Comment:
I think this should be a message that is pushed every time.  Instead of making the JS ask for the user location, just send it every time via onSendUserConfigLocation, just like onDataFilesRead is automatically pushed from C# to JS.

The save data is a little bit different because there's a "set" part of it where the pull counter sends back the current pull counts.  So, in that case, it's required that the setSaveData be an overlay message type that the JS can call back into C# with.

If I had to add some brief tweaks here, I'd also ask for:
* remove the automatic loading of user/whatever.js and use this path for that as well, so that there's only one way that user config files get loaded and remote files aren't a special case
* user location if unspecified should be ../../user relative to html file, so that this all works as it did prior to this pull request

I will mention offhand here that I think JS loading order is a bit muddled at the moment.  Currently, C# messages can be sent before the plugin is loaded.  (It'd be nice to delay sending anything but it wasn't clear how.  I'm not keen on JS having to inform C# when it's loaded.  Surely this can be automatic, somehow?)  So, many objects are set up, and then some time later, they're attached to the document to update elements.  This change adds an additional layer of loading weirdness where now user config is going to be loaded (likely) after the JS objects are created, so there may need to be some reinitialization in some cases.  I don't think this is something that needs to get fixed in this pull request, but it's something to keep a look out for.

Reply:
C# IIRC can get a callback from chromium when the page is loaded, but that doesn't always coincide with when javascript is ready.  It could happen too early or later than you want.

I was concerned about the options being changed after the settings as well, its one of the things I didn't get into testing.

I'll look more into it and make the tweaks you request.

Reply:
After looking at the system, the only way to ensure that there are no race conditions is to wait for the JS to send a request back.  I have pulled the code out into its own module and revised the code for raidboss, pullcounter, oopsyraidsy and jobs to use a callback pattern and registered events on DOMContentLoaded to make this work correctly and request the data at the earliest possible moment that there will be no issue of race condition.

I have also set everything up in such a way that Options are loaded, then user options will load, then the plugin will load, even though the user options are dependent on a callback, it will ensure those are loaded if needed, before the overlay instantiates the classes that use them.

I have some validation added to the userconfig file in case it doesn't exist, and otherwise if its blank it will load relative to the AppDomain's BaseDirectory, I'm still testing this to ensure it targets %ACT%\cactbot\user.